lib/mayaUsd/ufe/UsdUndoCreateGroupCommand.cpp

```diff
@@ -53,26 +74,59 @@ Ufe::SceneItem::Ptr UsdUndoCreateGroupCommand::group() const
 
 void UsdUndoCreateGroupCommand::undo()
 {
-	if (!fGroup) return;
+#if UFE_PREVIEW_VERSION_NUM < 2017	
+	if (!_group) return;
 
 	// See UsdUndoDuplicateCommand.undo() comments.
-	auto notification = Ufe::ObjectPreDelete(fGroup);
+	auto notification = Ufe::ObjectPreDelete(_group);
 	Ufe::Scene::notifyObjectDelete(notification);
 
-	auto prim = fGroup->prim();
+	auto prim = _group->prim();
 	auto stage = prim.GetStage();
 	auto usdPath = prim.GetPath();
 	stage->RemovePrim(usdPath);
 
-	fGroup.reset();
+	_group.reset();
+#else
+	for (UfeUndoCommands::reverse_iterator rit=_ufeUndoCommands.rbegin(); 
+			rit!=_ufeUndoCommands.rend(); ++rit) 
+	{
+		(*rit)->undo();
+	}
+#endif	
 }
 
 void UsdUndoCreateGroupCommand::redo()
 {
-	auto hierarchy = Ufe::Hierarchy::hierarchy(fParentItem);
+#if UFE_PREVIEW_VERSION_NUM < 2017	
+	auto hierarchy = Ufe::Hierarchy::hierarchy(_parentItem);
 	// See MAYA-92264: redo doesn't work.  PPT, 19-Nov-2018.
-	Ufe::SceneItem::Ptr group = hierarchy->createGroup(fName);
-	fGroup = std::dynamic_pointer_cast<UsdSceneItem>(group);
+	Ufe::SceneItem::Ptr group = hierarchy->createGroup(_name);
+	_group = std::dynamic_pointer_cast<UsdSceneItem>(group);
+#else
+	if (_ufeUndoCommands.empty()) {
+		auto addPrimCmd =  UsdUndoAddNewPrimCommand::create(_parentItem, _name.string(), "Xform");
+		_ufeUndoCommands.push_back(addPrimCmd);
+		addPrimCmd->redo();
+		
+		auto newParentSceneItem = Ufe::Hierarchy::createItem(addPrimCmd->newUfePath());
```

Comment:
This is nice, but since we're all in USD-land here, if we could just get the UsdPrim out of the addPrimCmd (and we know it's there), we could just call UsdSceneItem::create() directly, get a derived class UsdSceneItem out of it, and assign directly into _group, saving the dynamic_pointer_cast at line 122.  What do you think?  All you need is a prim accessor in UsdUndoAddNewPrimCommand, which sounds reasonable to me.